src/lib/sql/sql_pipeline.cpp

```diff
@@ -61,8 +62,14 @@ SQLPipeline::SQLPipeline(const std::string& sql, std::shared_ptr<TransactionCont
 
       parsed_statement->setIsValid(true);
 
-      auto pipeline_statement =
-          std::make_shared<SQLPipelineStatement>(std::move(parsed_statement), _transaction_context, use_mvcc);
+      // Get the statement string from the original query string, so we can pass it to the SQLPipelineStatement
+      DebugAssert(sql_string_offset < sql.length(),
+                  "SQL statements have a total length that is longer than the original string.");
+      const auto statement_string = boost::trim_copy(sql.substr(sql_string_offset, statement->stringLength));
+      sql_string_offset += statement->stringLength;
+
+      auto pipeline_statement = std::make_shared<SQLPipelineStatement>(statement_string, std::move(parsed_statement),
+                                                                       _transaction_context, use_mvcc);
       _sql_pipeline_statements.push_back(std::move(pipeline_statement));
     } catch (const std::exception&) {
```

Comment:
Why is this try/catch block necessary here?

Reply:
Because the `SQLParserResult` stores raw pointers of which we need to take ownership. So we release them and if something goes wrong then, the memory is never freed.

Reply:
Mhm okay, alternatively, you could wrap the pointers in smart pointers (probably unique_ptr in this case) when they are returned.